Rematch/Addon.lua

```diff
@@ -6,44 +6,335 @@ Addon.lua
+                end
+            end
+        end
+        local deleteItem = FindMenuItem(Rematch:GetMenu('TeamMenu'), DELETE)
+        self:RawHook(deleteItem, 'func', function(obj, key, ...)
+            self.hooks[deleteItem].func(obj, key, ...)
+
+            local origAccept = RematchDialog.acceptFunc
+            RematchDialog.acceptFunc = function(...)
+                self:RemoveScript(key)
+                return origAccept(...)
+            end
+        end, true)
+    end
+
+    -- Maintain sync between script name, and team name
+    if rematchVersion >= ns.Version:New(5, 0, 0, 0) then
+        local function overwriteName(self, teamID)
+            local team = Rematch.savedTeams[teamID]
+            local script = self:GetScript(teamID)
+
+            if team and script then
+                script:SetName(team.name)
+            end
+        end
+
+        Rematch.events:Register(self, 'REMATCH_TEAM_OVERWRITTEN', function(self, newTeamID, oldTeamID)
+            if not oldTeamID or
+               not newTeamID or
+               oldTeamID == newTeamID or
+               not self:GetScript(oldTeamID)then
+                return
+            end
+
+            self:MoveScript(oldTeamID, newTeamID)
+
+            overwriteName(self, teamID)
+        end)
+
+        Rematch.events:Register(self, 'REMATCH_TEAM_UPDATED', overwriteName)
+    else
+        local function rename(old, new)
+            if not old then
+                return
+            end
+            if old == new then
+                return
+            end
+            local script = self:GetScript(old)
+            if not script then
+                return
+            end
+
+            self:MoveScript(old, new)
+        end
+
+        local function errorhandler(err)
+            return geterrorhandler()(err)
+        end
+
+        local function safecall(func, ...)
+            return xpcall(func, errorhandler, ...)
+        end
+
+        self:RawHook(Rematch, 'SaveAsAccept', function(...)
+            safecall(function()
+                local team, key = Rematch:GetSideline()
+                if not self.savedRematchTeams[key] or not Rematch:SidelinePetsDifferentThan(key) then
+                    rename(Rematch:GetSidelineContext('originalKey'), key)
+                end
+            end)
+            return self.hooks[Rematch].SaveAsAccept(...)
+        end, true)
+
+        self:SecureHook(Rematch, 'OverwriteAccept', function()
+            safecall(function()
+                rename(Rematch:GetSidelineContext('originalKey'), select(2, Rematch:GetSideline()))
+            end)
+        end)
+    end
+
+    if rematchVersion >= ns.Version:New(5, 0, 0, 0) then
+        Rematch.events:Register(self, 'REMATCH_TEAM_CREATED', function(self, teamID)
+            self:OnImportContinuation(teamID)
+        end)
+    else
+        -- TODO: Should be done in Rematch4 as well, but apparently nobody noticed for years,
+        -- so probably not worth bothering.
```

Comment:
I'm fine with this TODO never being done.

Reply:
We can remove this TODO then.